scylla/src/transport/session.rs

```diff
@@ -1870,9 +1868,15 @@ impl Session {
     }
 
     pub async fn check_schema_agreement(&self) -> Result<bool, QueryError> {
-        let connections = self.cluster.get_working_connections().await?;
+        let cluster_data = self.get_cluster_data();
+        let connections_iter = cluster_data.iter_working_connections();
 
-        let handles = connections.iter().map(|c| c.fetch_schema_version());
+        let handles = connections_iter.map(|c| async {
+            match c {
+                Ok(c) => c.fetch_schema_version().await,
+                Err(err) => Err(err),
+            }
+        });
         let versions = try_join_all(handles).await?;
 
         let local_version: Uuid = versions[0];
```

Comment:
`versions` might be empty, this line might panic.

Reply:
I disagree. See the other comments.